modules/api/php/endpoints/candidate/visit/dicom/processes.class.inc

```diff
@@ -0,0 +1,255 @@
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('imaging_uploader')) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
+
+        try {
+            $mriuploads = $this->_visit->getMRIUploadsByFilename(
+                $user,
+                $this->_tarname
+            );
+        } catch (\NotFound $e) {
+            return new \LORIS\Http\Response\JSON\NotFound();
+        }
+
+        $array = (new \LORIS\api\Views\Visit\Dicom\Processes($mriuploads))
+            ->toArray();
+
+        $this->_cache = new \LORIS\Http\Response\JsonResponse($array);
+
+        return $this->_cache;
+    }
+
+    /**
+     * Launch the MRI pipeline on the DICOM study.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request): ResponseInterface
+    {
+        // *************************
+        // *** Permission checks ***
+        // *************************
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('imaging_uploader')) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
+
+        $data        = json_decode($request->getBody()->__toString(), true);
+        $processtype = $data['ProcessType'] ?? null;
+        if (!in_array($processtype, ['mri_upload'])) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Invalid ProcessType.'
+            );
+        }
+
+        $mriuploadid = $data['MriUploadID'] ?? null;
+        if (!is_numeric($mriuploadid)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Invalid ProcessType.'
+            );
+        }
+
+        try {
+            $mriupload = $this->_visit->getMRIUploadsById(
+                $user,
+                (int) $mriuploadid
+            );
+        } catch (\NotFound $e) {
+            return new \LORIS\Http\Response\JSON\NotFound(
+                'Filename and MRIUploadID do not match'
+            );
+        }
+
+        try {
+            $processid = (new ServerProcessLauncher())->mriupload(
+                $mriuploadid,
+                $mriupload->getUploadLocation()
+            );
+        } catch (\LorisException $e) {
+            error_log($e->getMessage());
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                'MRI upload process can not be launched (see error_log.)'
+            );
+        }
+
+        $ids = [$processid];
+        $serverProcessesMonitor = new ServerProcessesMonitor();
+        $processesState         = $serverProcessesMonitor->getProcessesState(
+            $ids,
+            $user->getUsername(),
+            $processtype
+        );
+
+        $lorispath = $request->getQueryParams()['lorispath'] ?? '';
+        $newpath   = $lorispath . '/' .$processid;
```

Comment:
Concatenating strings with user input is very risky. I don't think there's a direct attack possible the string is just going into a response, but it's possible to construct arbitrary urls that escape the module if an attacker knows what you're going to do with it.

1. Use Utility::pathJoin if you need to join path.
2. Don't assume lorispath is reliable, it comes from a get parameter in an htaccess rewrite. Use the BaseURL instead. Also you removed lorispath from the request object in a55cb25bf1daa167f0b1055a0ae272f67667a741

Reply:
Whoops, this was old code I forgot to update and it seems there was indeed a bug here. Fixed in this commit:
https://github.com/aces/Loris/pull/9154/commits/bfb2873811046cdd4e1c5b5b1550b4800fbce82d
(it still uses concatenation but that should be enough IMO)